tests/bsim/bluetooth/host/gatt/sc_indicate/src/central.c

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(test_central, LOG_LEVEL_DBG);
+
+#include "bs_bt_utils.h"
+
+DEFINE_FLAG(flag_discovered);
+DEFINE_FLAG(flag_subscribed);
+DEFINE_FLAG(flag_indicated);
+
+enum GATT_HANDLES {
+	SC,
+	CCC,
+	NUM_HANDLES,
+};
+
+static uint16_t gatt_handles[NUM_HANDLES] = {0};
+
+static struct bt_gatt_subscribe_params subscribe_params;
+
+static void sc_subscribed(struct bt_conn *conn,
+			  uint8_t err,
+			  struct bt_gatt_subscribe_params *params)
+{
+	LOG_DBG("subscribed");
+	SET_FLAG(flag_subscribed);
+}
+
+static uint8_t sc_indicated(struct bt_conn *conn,
+			    struct bt_gatt_subscribe_params *params,
+			    const void *data,
+			    uint16_t length)
+{
+	LOG_DBG("indication received");
+
+	SET_FLAG(flag_indicated);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static void subscribe(void)
+{
+	int err;
+
+	subscribe_params.ccc_handle = gatt_handles[CCC];
+	subscribe_params.value_handle = gatt_handles[SC];
+	subscribe_params.value = BT_GATT_CCC_INDICATE;
+	subscribe_params.subscribe = sc_subscribed;
+	subscribe_params.notify = sc_indicated;
+
+	err = bt_gatt_subscribe(get_g_conn(), &subscribe_params);
+	BSIM_ASSERT(!err, "bt_gatt_subscribe failed (%d)\n", err);
+
+	WAIT_FOR_FLAG(flag_subscribed);
+}
+
+static uint8_t discover_func(struct bt_conn *conn,
+			     const struct bt_gatt_attr *attr,
+			     struct bt_gatt_discover_params *params)
+{
+	if (attr == NULL) {
+		for (int i = 0; i < ARRAY_SIZE(gatt_handles); i++) {
+			LOG_DBG("handle[%d] = 0x%x", i, gatt_handles[i]);
+			BSIM_ASSERT(gatt_handles[i] != 0, "did not find all handles\n");
+		}
+		(void)memset(params, 0, sizeof(*params));
+		SET_FLAG(flag_discovered);
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	if (params->type == BT_GATT_DISCOVER_CHARACTERISTIC) {
+		const struct bt_gatt_chrc *chrc = (struct bt_gatt_chrc *)attr->user_data;
+		const struct bt_uuid_16 ccc_uuid = BT_UUID_INIT_16(BT_UUID_GATT_CCC_VAL);
+
+		if (bt_uuid_cmp(chrc->uuid, BT_UUID_GATT_SC) == 0) {
+			int err;
+
+			LOG_DBG("found sc");
+			gatt_handles[SC] = chrc->value_handle;
+
+			params->uuid = &ccc_uuid.uuid;
```

Comment:
`ccc_uuid` is a local variable, but the `params` supplied to `bt_gatt_discover` should be valid until it completes, so suggest to make it a static